openlibrary/tests/core/test_i18n.py

```diff
@@ -13,24 +15,40 @@ def ungettext(self, message1, message2, n):
         else:
             return self.gettext(message2)
 
+
 class MockLoadTranslations(dict):
     def __call__(self, lang):
         return self.get(lang)
 
     def init(self, lang, translations):
         self[lang] = MockTranslations(translations)
 
+
 class Test_ungettext:
     def setup_monkeypatch(self, monkeypatch):
         self.d = MockLoadTranslations()
         ctx = web.storage()
 
         monkeypatch.setattr(i18n, "load_translations", self.d)
         monkeypatch.setattr(web, "ctx", ctx)
+        self._load_fake_context()
```

Comment:
Some tests of the HTTP Accept-Language header are needed, including ones that request languages which aren't supported or where the first supported language is 2nd or 3rd on the list.

Reply:
Ok. Yes, I will try to provide tests for the new solution.

Reply:
New tests for the `Accept-Language` header will be provided (including the tests you've mentioned in your comment).